laboratory-service/src/main/java/com/raf/si/laboratoryservice/service/impl/ReferralServiceImpl.java

```diff
@@ -86,17 +92,75 @@ public ReferralResponse deleteReferral(Long id) {
+
         List<Referral> unprocessedReferrals = unprocessedReferralsByThreeParams.stream()
                 .filter(referral -> referral.getLabWorkOrder() == null)
                 .collect(Collectors.toList());
 
-        return referralMapper.referralListToListResponse(unprocessedReferrals);
+        List<UnprocessedReferralsResponse> unprocessedReferralsResponses = new ArrayList<>();
+        List<DepartmentResponse> allDepartments = getDepartments(authorizationHeader);
+        List<DoctorResponse> allDoctors = getAllDoctors(authorizationHeader);
+
+        for (Referral referral : unprocessedReferrals) {
+            UnprocessedReferralsResponse unprocessedReferral = new UnprocessedReferralsResponse();
+            unprocessedReferral.setReferralId(referral.getId());
+
+            for (DepartmentResponse departmentResponse: allDepartments) {
+                if (referral.getPboReferredFrom().equals(departmentResponse.getPbo())) {
+                    unprocessedReferral.setDepartmentName(departmentResponse.getName());
+                }
+            }
+
+            for (DoctorResponse doctor: allDoctors) {
+                if (referral.getLbz().equals(doctor.getLbz())) {
+                    unprocessedReferral.setDoctorFirstName(doctor.getFirstName());
+                    unprocessedReferral.setDoctorLastName(doctor.getLastName());
+                }
+            }
+            unprocessedReferral.setRequiredAnalysis(referral.getRequiredAnalysis());
+            unprocessedReferral.setCreationDate(referral.getCreationTime());
+            unprocessedReferral.setComment(referral.getComment());
+            unprocessedReferralsResponses.add(unprocessedReferral);
+        }
+
+        return unprocessedReferralsResponses;
+    }
+
+    private List<DoctorResponse> getAllDoctors(String token) {
+        ResponseEntity<List<DoctorResponse>> response;
+        List<DoctorResponse> responseBody;
+        try {
+            response = HttpUtils.getAllDoctors(token);
+            responseBody = response.getBody();
+
+        } catch (IllegalArgumentException e) {
+            String errMessage = String.format("Error when calling user service: " + e.getMessage());
+            log.info(errMessage);
+            throw new InternalServerErrorException("Error when calling user service: " + e.getMessage());
+        }
+        return responseBody;
+    }
+
+    private List<DepartmentResponse> getDepartments(String token) {
+        ResponseEntity<List<DepartmentResponse>> response;
+        List<DepartmentResponse> responseBody;
+        try {
+            response = HttpUtils.findDepartmentName(token);
+            responseBody = response.getBody();
+
+        } catch (IllegalArgumentException e) {
+            String errMessage = String.format("Error when calling user service: " + e.getMessage());
+            log.info(errMessage);
+            throw new InternalServerErrorException("Error when calling user service: " + e.getMessage());
+        }
```

Comment:
Mozda treba proveriti ako se vraca odredjeni status code koji je izmedju 400 i 500 da se propagira takav error?

Reply:
Vazi, dodajem to